### PR TITLE
Replace jshint with eslint. Closes #596 and closes #601.

### LINKED ISSUE #596: Linters
the simple long loved jscs is being end of life'd and consolidating with eslint. this project is using dockyard's ember-suave. they've started working on an [eslint port](https://github.com/DockYard/eslint-plugin-ember-suave) of it; it's still experimental but worth trying. 

**Expected behavior:**  keeping the code style focused and unified

**Actual behavior:** deprecation warnings

**Steps to reproduce:** being dumb enough (like me) to update your global installation of jscs.

**OS and Browser:** os x 10.11.5 and chrome (atom/vim)

### PR DESCRIPTION
**Pull request by btecu**
`jshint` has merged with `eslint` and only `eslint` will be supported.

Note: Using `ember-cli-eslint` version `2.0.0` because the latest `3.0.0` has dropped support for node `0.10` and `0.12`.

Will follow additional PR to fix some newly discovered warnings.

cc @HospitalRun/core-maintainers

### CODE REVIEW

**Comment by btecu** [comment]
Note that `eslint-plugin-ember-suave` is included but not turned on.
I plan on submitting another pull request where I turn it on and fix all the errors.

**Comment by jkleinsc** [comment]
@btecu we don't need to support node 0.10 and 0.12, so I would recommend using the latest ember-cli-eslint

**Comment by btecu** [comment]
@jkleinsc Great, I've updated `ember-cli-eslint` to version `3.0.0`.

File: package.json [hunk 1]
```diff
@@ -73,6 +73,7 @@
     "ember-suave": "4.0.1",
```
Comment (btecu) [inline package.json]: Good catch. Updated.

**Pull request merged**